guide.md: added consistency to 04-data-types-and-format section (#342)
datacarpentry/python-ecology-lesson: _extras/guide.md

### Challenge - Changing Types

> * Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Next try converting `weight` to an integer. What goes wrong here? What is Pandas telling you?

Reviewer: Next -> then

Reviewer: `Then try converting `weight` column to an integer type`

Reviewer: I'd remove `What goes wrong here?`... and say `What does _pandas_ tell you when you try doing that?` but I'll leave it to @wrightaprilm to decide what is the best wording here

Reviewer 2: How about "What error message does Pandas give you? What does this error mean?" Emphasizing specifically that we want them to read and comprehend the output, which in this case is very helpful and informative!

Author: Thanks @maxim-belkin  and @wrightaprilm. Is this a good wording to commit?

```suggestion
* Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Then, try converting the contents of the `weight` column to an integer type. What error message does Pandas give you? What does this error mean?
```

Reviewer 2: I like this wording!

> * Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Then, try converting the contents of the `weight` column to an integer type. What error message does Pandas give you? What does this error mean?

Author: Already committed those changes. Thank you for pointing out @maxim-belkin!

Pandas cannot convert types from float to int if the column contains NaN values.

### Challenge - Counting

> * Count the number of missing values per column. Hint: The method .count() gives you the number of non-NA observations per column. Try looking to the .isnull() method.

Reviewer: use backticks around .count() and .isnull() so that they appear as `.count()` and `.isnull()`

Reviewer 2: ```suggestion
* Count the number of missing values per column. Hint: The method `.count()` gives you the number of non-NA observations per column. Try looking to the `.isnull()` method.
```

If you just want to accept this change, @NPalopoli, it contains the correct change.

~~~
surveys_df.isnull()
~~~
{: .language-python}

> If the students have trouble generating the output, or anything happens with that, there is a file called "sample output" that contains the data file they should generate.

Reviewer 2: "there is a folder called `sample_output`, which contains the data file they should generate. It is not included with the data in the setup steps."

Most instructors don't have the learners clone the repo, as Git and GitHub are not part of the Data Carpentry core. So learners typically don't have this file, but the instructor could provide it.

Author: Nice call April. Perhaps this is a good wording?

```suggestion
If the students have trouble generating the output, or anything happens with that, there is a file called "sample output" that contains the data file they should generate. It is not included with the data in the setup steps, so remember to have it at hand and provide it if needed.
```

Reviewer: @NPalopoli, note "folder" vs "file".
Another note: "there is" does not answer the question of "where". I suggest saying something like `This repository has a folder called...`

Author: Thank you @maxim-belkin once and again! Should be correct now.
